lib/view_component/slot_v2.rb

```diff
@@ -22,8 +22,10 @@ def initialize(parent)
     # If there is no slot renderable, we evaluate the block passed to
     # the slot and return it.
     def to_s
+      return @content if defined?(@content)
```

Comment:
I think this makes this problem more complex, but I don't think we want to memoize here. I don't believe components memoize their output, so I'm a little hesitant about memoizing slots output too.

Reply:
the reason I did this is because `html_safe?` has to call `to_s` and later on temple also calls `to_s` so I wanted to avoid the second "render"

Reply:
Yep, makes sense.

I've been thinking on it a bit this morning, and it's probably fine. I'm not sure we'll run into the use-case of 'rendering a slot once, changing some properties, then rendering it again' very often. So this is probably an acceptable trade-off. 👍